src/AutoRest.CSharp/DataPlane/Output/DataPlaneRestClient.cs

```diff
@@ -1,23 +1,74 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Common.Output.Builders;
 using AutoRest.CSharp.Input;
 using AutoRest.CSharp.Output.Models.Requests;
 using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Utilities;
 
 namespace AutoRest.CSharp.Output.Models
 {
     internal class DataPlaneRestClient : RestClient
     {
         private BuildContext<DataPlaneOutputLibrary> _context;
 
+        public IReadOnlyList<LowLevelClientMethod>? ProtocolMethods => _context.Configuration.ProtocolMethodList.Length > 0 && GetProtocolMethods() != null
```

Comment:
Do we always have only one `*RestClient` per project?

Reply:
No, it can be more than one RestClient per project and with this change if we have "Delete" `OperationId` in 2 RestClients, it will generate protocol method for both of them. Since the protocol method generated internally and we want to make it simple, we are taking only operationId for now. In future if it will create a lot of noise in RestClient, we will change and take key, value pair(<`OperationGroup`, List<<`OperationId`>>>).

Reply:
Honestly, I'd prefer this all to be configured in partial types. But we can discuss it when/if we need more granulated configuration.

Reply:
Yes, but in partial RestClient type, devs would have to write protocol method manually. We added this config, so that this method will be auto generated.

Reply:
We can have that discussion when we get there -- I'd like to understand how it would work in partial types, and what that looks like.  I think the solution we have is good for the current problem we're solving.